lib/upperkut/batch_execution.rb

```diff
@@ -13,29 +13,25 @@ def execute
       worker_instance = @worker.new
       items = @worker.fetch_items.freeze
 
-      items_body = items.collect do |item|
-        item['body']
-      end
-
       @worker.server_middlewares.invoke(@worker, items) do
-        worker_instance.perform(items_body.dup)
+        items_body = items.map(&:body)
+        worker_instance.perform(items_body)
       end
     rescue StandardError => error
       @logger.info(
         action: :requeue,
         ex: error,
-        item_size: items_body.size
+        item_size: items.size
       )
 
       @logger.error(error.backtrace.join("\n"))
 
       if worker_instance.respond_to?(:handle_error)
-        worker_instance.handle_error(error, items_body)
+        worker_instance.handle_error(error, items)
```

Comment:
:+1:

Reply:
chegou avaliar se quebra compatibilidade coma versão passada?

Reply:
quem implementou handle_error em workers na versão passada, precisa alterar algo para essa versão?

Reply:
Depende do que eles fazem com o Item. Se só acessam alguma informação usando interface de Hash ou se executam alguma lógica boba e re-agendam, não quebrou, mas se for algo um pouco mais complexo pode quebrar sim.

Reply:
estou verificando se existe algum handle_error existe na org.

Reply:
[Aqui](https://github.com/ResultadosDigitais/clustering-engine/blob/master/app/workers/cluster_test_request_worker.rb#L17) não quebraria, por exemplo.